core/Pos/Core/Block.hs

```diff
@@ -75,6 +77,9 @@ class Blockchain p where
     default checkBodyProof :: Eq (BodyProof p) => Body p -> BodyProof p -> Bool
     checkBodyProof body proof = mkBodyProof body == proof
 
+-- | Extension of 'Blockchain' type class with helper functions.
+class Blockchain p => BlockchainHelpers p where
+    verifyBBlockHeader :: MonadError Text m => GenericBlockHeader p -> m ()
```

Comment:
Add description to these -- which checks are performed. Because it does obviously less than `Pure.verifyX` can.